src/globus_cli/parsing/shared_options/endpointish.py

```diff
@@ -0,0 +1,319 @@
+                type=StringOrNull(),
+            ),
+            click.option(
+                "--contact-info",
+                help=f"Contact info for the {name}",
+                type=StringOrNull(),
+            ),
+            click.option(
+                "--contact-email",
+                help=f"Contact email for the {name}",
+                type=StringOrNull(),
+            ),
+            click.option(
+                "--organization",
+                help=f"Organization for the {name}",
+                type=StringOrNull(),
+            ),
+            click.option(
+                "--department",
+                help=f"Department which operates the {name}",
+                type=StringOrNull(),
+            ),
+            click.option(
+                "--keywords",
+                type=str if keyword_style == "string" else CommaDelimitedList(),
+                help="Comma separated list of keywords to help searches "
+                f"for the {name}",
+            ),
+            click.option(
+                "--default-directory",
+                type=StringOrNull(),
+                help="Default directory when browsing or executing tasks "
+                f"on the {name}",
+            ),
+            click.option(
+                "--force-encryption/--no-force-encryption",
+                default=None,
+                help=f"Force the {name} to encrypt transfers",
+            ),
+        ]
+    )
+
+    if verify_style == "choice":
+        verify_help = (
+            f"Set the policy for this {name} for file integrity verification "
+            "after transfer. 'force' requires all transfers to perform "
+            "verfication. 'disable' disables all verification checks. 'default' "
+            "allows the user to decide on verification at Transfer task submit  "
+            "time."
+        )
+        if name == "collection":
+            verify_help += (
+                " When set on mapped collections, this policy is inherited by any "
+                "guest collections"
+            )
+        decorators.append(
+            click.option(
+                "--verify",
+                type=click.Choice(
+                    ["force", "disable", "default"], case_sensitive=False
+                ),
+                callback=_verify_choice_to_dict,
+                help=verify_help,
+                type_annotation=DictType[str, bool],
+                cls=AnnotatedOption,
+            )
+        )
+    else:
+        decorators.append(
+            click.option(
+                "--disable-verify/--no-disable-verify",
+                default=None,
+                is_flag=True,
+                help=f"Set the {name} to ignore checksum verification",
+            )
+        )
+
+    return utils.fold_decorators(f, decorators)
+
+
+def _verify_choice_to_dict(
+    ctx: click.Context, param: click.Parameter, value: t.Any
+) -> dict[str, bool]:
+    if value is None:
+        return {}
```

Comment:
Should None be treated as default instead of null?

Reply:
I'm not sure I follow the question.

To explain how I see it:

The callback runs even if no option is passed. So this ensures that we can write
```python
@decorator_here
def foo(..., verify: dict[str, bool]): ...
```
If we made it `None`, then the functions would change
```python
@decorator_here
def foo(..., verify: dict[str, bool] | None): ...
```

And the way that `verify` is being used is in usages like `dict(foo=foo, bar=bar, **verify)`.

We could handle this option a lot of different ways -- I'm not overly attached to what's here -- but I'm not sure what you're proposing as the replacement logic.

Reply:
Ah that makes sense. I was missing the invocation scope on this one, thanks for explaining